commit b361ee938f56f679a1b09dafdc0cc17485f5ec30 (refs/changes/04/3204/2)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2018-01-12T01:27:20+00:00 (1 year, 9 months ago)
    
    MANTA-3547 muskie fails assertion on empty request body for mpu-create

diff --git a/lib/uploads/commit.js b/lib/uploads/commit.js
index 5fb63a9..08f737b 100644
--- a/lib/uploads/commit.js
+++ b/lib/uploads/commit.js
@@ -35,11 +35,14 @@ var hasKey = jsprim.hasKey;
 var shallowCopy = utils.shallowCopy;
 var sprintf = util.format;
 var schemaValidator = ajv.compile({
+    'type': 'object',
     'properties': {
         'parts': {
             'type': 'array',
+            'maxItems': uploadsCommon.MAX_NUM_PARTS,
             'items': {
-                'type': 'string'
+                'type': 'string',
+                'minLength': 1
             }
         }
     },
@@ -235,18 +238,6 @@ function validateParts(req, res, next) {
     var parts = req.body.parts;
     log.debug('validating parts for upload');
 
-    if (!parts) {
-        log.debug('undefined parts array');
-        next(new MultipartUploadCreateError('missing parts array in ' +
-                    'request body'));
-        return;
-    } else if (parts.length > uploadsCommon.MAX_NUM_PARTS) {
-        next(new MultipartUploadInvalidArgumentError(id,
-            sprintf('too many parts provided (%d provided; max is %d)',
-                parts.length, uploadsCommon.MAX_NUM_PARTS)));
-        return;
-    }
-
     var errors = [];
     var sum = 0;
 
@@ -264,13 +255,6 @@ function validateParts(req, res, next) {
         var record = req.upload.uploadMd;
         var key = record.key + '/' + index;
 
-        if (etag === '') {
-            errors.push(new MultipartUploadInvalidArgumentError(id,
-                sprintf('[part %d] no etag provided', index, etag)));
-            cb();
-            return;
-        }
-
         var opts = {
             key: key,
             requestId: req.getId()
diff --git a/lib/uploads/create.js b/lib/uploads/create.js
index f034135..9f7b46c 100644
--- a/lib/uploads/create.js
+++ b/lib/uploads/create.js
@@ -38,9 +38,11 @@ require('../errors');
 var hasKey = jsprim.hasKey;
 var sprintf = util.format;
 var schemaValidator = ajv.compile({
+    'type': 'object',
     'properties': {
         'objectPath': {
-            'type': 'string'
+            'type': 'string',
+            'minLength': 1
         },
         'headers': {
             'type': 'object',
